lib/_nonstd.scm

```diff
@@ -3067,6 +3067,17 @@
 (define-prim (executable-path)
   (##executable-path))
 
+(define-prim (##process-mnemonic-identifier)
+  (let ((path (if (##pair? ##processed-command-line)
```

Comment:
There's a subtle bug here. The code is not thread safe. Another thread could change `##process-mnemonic-identifier` between the test for pair and the call to car. Also, the PID should be added to the name in all cases. So this definition is better:
```
(define-primitive (##process-mnemonic-identifier)
  (let* ((pcl
          ##processed-command-line)
         (path
          (if (##pair? pcl)
              (##car pcl)
              (let ((exec-path (##os-executable-path)))
                (if (##string? exec-path)
                    exec-path
                    #f)))))
    (##string-append
     (if path
         (##path-strip-extension (##path-strip-directory path))
         "process")
     "-"
     (##number->string (##os-getpid)))))
```

Reply:
Got it -- thanks. (Note for myself: in the fixed version, binding `pcl` lets us write the procedure with only one reference to the free variable `##processed-command-line`.)